hcloud/resource_hcloud_reversedns.go

```diff
@@ -0,0 +1,240 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"log"
+	"net"
+	"strings"
+)
+
+func resourceReverseDns() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDnsRead,
+		Create: resourceReverseDnsCreate,
+		Delete: resourceReverseDnsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns_ptr": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceReverseDnsRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ctx := context.Background()
+
+	ipAddress := string(d.Get("ip_address").(string))
+	rdnsType := ""
+	id, ok := d.GetOk("server_id")
```

Comment:
Please lookup the server/floating ip via the assigned record id.
You could use a function like this:

```go
func lookupRDNSID(id string) (server *hcloud.Server, floatingIP *hcloud.FloatingIP, err error) {
	// lookup code
	return
}
```
and use it like this:
```go
server, floatingIP, err = lookupRDNSID(id)
if err != nil {
	if err == errInvalidRDNSID {
		// delete from state
	}
	return err
}
if server != nil {
	// handle server
}
// floatingIP must be != nil here because otherwise 'lookupRDNSID' would,
// have returned an error "not found" or "invalid id"
```